python/ambassador/ir/irhost.py

```diff
@@ -473,7 +478,41 @@ def matches_httpgroup(self, group: "IRHTTPMappingGroup") -> bool:
                 sel_match,
             )
 
-        return host_match or sel_match
+        groupName = group.get("name") or "None"
+
+        # The synthetic Mappings for diagnostics, readiness, and liveness probes always match all Hosts.
+        # They can all still be disabled if desired via the Ambassador Module resource
+        if groupName in [
+            "GROUP: internal_readiness_probe_mapping",
+            "GROUP: internal_liveness_probe_mapping",
+            "GROUP: internal_diagnostics_probe_mapping",
+        ]:
+            return True
```

Comment:
Relying on the internal group name feels slimy to me. But, uh, I guess it's the best we have.

Reply:
I agree that it does feel weird, but I couldn't think of a way to use a special field that seemed better. Are you suggesting using `ambassador_id` itself, or adding a new field?

If the group names were ever to change, the tests will fail so that at least makes me feel a little better about using them.

Reply:
I was suggesting adding a new field, or otherwise using a special sentinel value in an existing field.

Good point about the test.